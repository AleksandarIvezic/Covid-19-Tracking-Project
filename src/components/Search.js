import React from 'react';
import PropTypes from 'prop-types';
import '../styling/Search.css';

const Search = ({ handleChange }) => (
  <form className="d-flex stripe-text justify-content-between">
    <span className="m-1">STATS BY COUNTRY</span>
    <input className="search" name="search" type="text" placeholder="Search" onChange={handleChange} />
  </form>
);

Search.propTypes = {
  handleChange: PropTypes.func.isRequired,
};

export default Search;
