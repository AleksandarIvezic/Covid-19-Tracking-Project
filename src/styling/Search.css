.stripe-text {
  font-size: 0.7rem;
  background-color: #35548b;
}

.search {
  width: 45%;
}
