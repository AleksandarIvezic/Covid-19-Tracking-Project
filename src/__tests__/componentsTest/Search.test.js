import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/storeConfig';
import Search from '../../components/Search';

describe('Search', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Search handleChange={() => 2 + 3} /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
