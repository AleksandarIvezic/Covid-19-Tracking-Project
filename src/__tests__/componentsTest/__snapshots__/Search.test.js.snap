// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<form
  className="d-flex stripe-text justify-content-between"
>
  <span
    className="m-1"
  >
    STATS BY COUNTRY
  </span>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
</form>
`;
